### PR TITLE
Microsoft.Data.Sqlite: Add Default Timeout connection string keyword

### LINKED ISSUE #22505: Microsoft.Data.Sqlite: Add Default Timeout connection string keyword
We should add the `Default Timeout` connection string keyword. Order of precedence would be:

1. SqliteCommand.CommandTimeout (always wins if set)
2. SqliteConnection.DefaultTimeout
3. `Default Time=` in the connection string
4. 30 (the ADO.NET default)

### PR DESCRIPTION
**Pull request by nmichels**
- Fixes #22505

### CODE REVIEW

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs [hunk 1]
```diff
@@ -275,6 +275,11 @@ public override void Open()
                     this.ExecuteNonQuery("PRAGMA recursive_triggers = 1;");
                 }
 
+                if (ConnectionOptions.DefaultTimeout.HasValue)
+                {
+                    DefaultTimeout = ConnectionOptions.DefaultTimeout.Value;
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs:280]: I think that if the user sets SqliteConnection.DefaultTimeout, it should take precedence. Only fallback to the connection string value when it’s unset.

**Review by bricelam — CHANGES_REQUESTED** [review]
LGTM

File: src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs (hunk 1 above)
Comment (bricelam) [reply src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs:280]: Also, do we ever use this value when the connection is closed? If so, we may need to read the value in the ConnectionString property setter.

Comment (nmichels) [reply src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs:280]: Hi @bricelam 
By unset you mean different from the default value (30)? Or should we implement a private property and explicitly check wheter the DefaultTimeout has changed in it's setter?
I'm not sure, but I think that it doesn't make sense to use this property value when the connection is closed.
Thanks!

Comment (nmichels) [reply src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs:280]: In fact, if we move this code snippet to the ConnectionString property setter we will have the desired precedence.

**Review by bricelam — APPROVED** [review]
I reworked this a bit while rebasing since the underlying code changed in PR #23549

**Review by smitpatel — APPROVED** [review]

**Pull request merged**